tools/etcd-dump-logs/etcd-dump-log_test.go

```diff
@@ -0,0 +1,277 @@
+	"bytes"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/coreos/etcd/auth/authpb"
+	"github.com/coreos/etcd/etcdserver/etcdserverpb"
+	"github.com/coreos/etcd/pkg/fileutil"
+	"github.com/coreos/etcd/pkg/pbutil"
+	"github.com/coreos/etcd/raft/raftpb"
+	"github.com/coreos/etcd/wal"
+	"go.uber.org/zap"
+)
+
+func TestEtcdDumpLogEntryType(t *testing.T) {
+	// directory where the command is
+	binDir, err := os.Getwd()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	dumpLogsBinary := path.Join(binDir + "/etcd-dump-logs")
+	if !fileutil.Exist(dumpLogsBinary) {
+		t.Skipf("%q does not exist", dumpLogsBinary)
+	}
+
+	p, err := ioutil.TempDir(os.TempDir(), "etcddumplogstest")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(p)
+
+	memberdir := filepath.Join(p, "member")
+	err = os.Mkdir(memberdir, 0744)
+	if err != nil {
+		t.Fatal(err)
+	}
+	waldir := walDir(p)
+	snapdir := snapDir(p)
+
+	w, err := wal.Create(zap.NewExample(), waldir, nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = os.Mkdir(snapdir, 0744)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	ents := make([]raftpb.Entry, 0)
+
+	// append entries into wal log
+	appendConfigChangeEnts(&ents)
+	appendNormalRequestEnts(&ents)
+	appendNormalIRREnts(&ents)
+	appendUnknownNormalEnts(&ents)
+
+	// force commit newly appended entries
+	err = w.Save(raftpb.HardState{}, ents)
+	if err != nil {
+		t.Fatal(err)
+	}
+	w.Close()
+
+	argtests := []struct {
+		name         string
+		args         []string
+		fileExpected string
+	}{
+		{"no entry-type", []string{p}, "expectedoutput/listAll.output"},
+		{"confchange entry-type", []string{"-entry-type", "ConfigChange", p}, "expectedoutput/listConfigChange.output"},
+		{"normal entry-type", []string{"-entry-type", "Normal", p}, "expectedoutput/listNormal.output"},
+		{"request entry-type", []string{"-entry-type", "Request", p}, "expectedoutput/listRequest.output"},
+		{"internalRaftRequest entry-type", []string{"-entry-type", "InternalRaftRequest", p}, "expectedoutput/listInternalRaftRequest.output"},
+		{"range entry-type", []string{"-entry-type", "IRRRange", p}, "expectedoutput/listIRRRange.output"},
+		{"put entry-type", []string{"-entry-type", "IRRPut", p}, "expectedoutput/listIRRPut.output"},
+		{"del entry-type", []string{"-entry-type", "IRRDeleteRange", p}, "expectedoutput/listIRRDeleteRange.output"},
+		{"txn entry-type", []string{"-entry-type", "IRRTxn", p}, "expectedoutput/listIRRTxn.output"},
+		{"compaction entry-type", []string{"-entry-type", "IRRCompaction", p}, "expectedoutput/listIRRCompaction.output"},
+		{"lease grant entry-type", []string{"-entry-type", "IRRLeaseGrant", p}, "expectedoutput/listIRRLeaseGrant.output"},
+		{"lease revoke entry-type", []string{"-entry-type", "IRRLeaseRevoke", p}, "expectedoutput/listIRRLeaseRevoke.output"},
+		{"confchange and txn entry-type", []string{"-entry-type", "ConfigChange,IRRCompaction", p}, "expectedoutput/listConfigChangeIRRCompaction.output"},
+	}
```

Comment:
@gyuho 
1. Thank you for catching the regression that when no entry-type is defined, the tool did not list all the entries. Now I have fixed that and added all types of InternalRaftRequest and made sure that no entries are missing in cases like: 1. no entry-type defined, 2. -entry-type Normal, 3. -entry-type InternalRaftRequest. 

2. I added filters for compaction, leasegrant and leaserevoke types and added tests for those.